File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
import heapq

def solution(sizes):
    n = len(sizes)
    
    for i in range(n):
        w = sizes[i][0]
        h = sizes[i][1]
        if w < h:
            sizes[i][0] = h
            sizes[i][1] = w
    a = []
    b = []
    for w ,h  in sizes:
        a.append(w)
        b.append(h)
    w = max(a)
    h = max(b)

    return w*h

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(int[,] sizes) {
        int maxX = 0;
        int maxY = 0;
        
        int n = sizes.GetLength(0);
        Console.Write(n);
        
        for(int i = 0; i<n; i++){
            int a = sizes[i,0];
            int b = sizes[i,1];
            if( b > a){
                sizes[i,0] = b;
                sizes[i,1] = a;
            }
            if(sizes[i,0] > maxX) maxX = sizes[i,0];
            if(sizes[i,1] > maxY) maxY = sizes[i,1];
        }
        
        return maxX*maxY;
    }
}